tests/test_utils.py

```diff
@@ -0,0 +1,89 @@
+from _pytest.logging import LogCaptureFixture
+import pytest
+
+from pytest_mock import MockerFixture
+
+from mock import Mock
+
+import dns.resolver
+import dns.rdatatype
+import dns.rdataclass
+import dns.message
+import dns.rrset
+import dns.flags
+
+from pythclient.utils import get_key
+
+
+@pytest.fixture()
+def get_answer_program() -> dns.resolver.Answer:
+    qname = dns.name.Name(labels=(b'devnet-program-v2', b'pyth', b'network', b''))
+    rdtype = dns.rdatatype.TXT
+    rdclass = dns.rdataclass.IN
+    response = dns.message.QueryMessage(id=0)
+    response.flags = dns.flags.QR
+    rrset_qn = dns.rrset.from_text(qname, 100, rdclass, rdtype)
+    rrset_ans = dns.rrset.from_text(qname, 100, rdclass, rdtype, '"program=gSbePebfvPy7tRqimPoVecS2UsBvYv46ynrzWocc92s"')
+    response.question = [rrset_qn]
+    response.answer = [rrset_ans]
+    answer = dns.resolver.Answer(
+        qname=qname, rdtype=rdtype, rdclass=rdclass, response=response)
+    answer.rrset = rrset_ans
+    return answer
+
+
+@pytest.fixture()
+def get_answer_mapping() -> dns.resolver.Answer:
+    qname = dns.name.Name(labels=(b'devnet-mapping-v2', b'pyth', b'network', b''))
+    rdtype = dns.rdatatype.TXT
+    rdclass = dns.rdataclass.IN
+    response = dns.message.QueryMessage(id=0)
+    response.flags = dns.flags.QR
+    rrset_qn = dns.rrset.from_text(qname, 100, rdclass, rdtype)
+    rrset_ans = dns.rrset.from_text(qname, 100, rdclass, rdtype, '"mapping=BmA9Z6FjioHJPpjT39QazZyhDRUdZy2ezwx4GiDdE2u2"')
+    response.question = [rrset_qn]
+    response.answer = [rrset_ans]
+    answer = dns.resolver.Answer(
+        qname=qname, rdtype=rdtype, rdclass=rdclass, response=response)
+    answer.rrset = rrset_ans
+    return answer
+
+
+@pytest.fixture()
+def mock_dns_resolver_resolve(mocker: MockerFixture) -> Mock:
+    mock = Mock()
+    mocker.patch('dns.resolver.resolve', side_effect=mock)
+    return mock
+
+
+def test_utils_get_program_key(mock_dns_resolver_resolve: Mock, get_answer_program: dns.resolver.Answer) -> None:
+    mock_dns_resolver_resolve.return_value = get_answer_program
+    program_key = get_key("devnet", "program")
+    assert program_key == "gSbePebfvPy7tRqimPoVecS2UsBvYv46ynrzWocc92s"
```

Comment:
Data only and very little to no logic. This is a perfect example of a simple and good unit test. Nice 👍